cnd/src/http_api/routes.rs

```diff
@@ -99,25 +99,73 @@ pub async fn handle_get_halight_swap(
                 state.fund_action().map(|_| "fund"),
                 state.redeem_action().map(|_| "redeem"),
             ];
-
-            make_swap_entity(swap_id, maybe_action_names)
+            make_swap_entity(swap_id, state, maybe_action_names)
         }
-    };
-
-    Ok(entity)
+    }
 }
 
-fn make_swap_entity(swap_id: LocalSwapId, maybe_action_names: Vec<Option<&str>>) -> siren::Entity {
-    let swap = siren::Entity::default().with_class_member("swap");
+// TODO: The API we are attempting to fulfil is defined in
+//   `comit-js-sdk/src/cnd/swaps_payload.ts`
+fn make_swap_entity<S>(
+    swap_id: LocalSwapId,
+    state: S,
+    maybe_action_names: Vec<Option<&str>>,
+) -> anyhow::Result<siren::Entity>
+where
+    S: Status + MyRole,
+{
+    let swap = SwapResource {
+        status: state.status(),
+        role: (), // FIXME: use state.role() once serialization fore Role is done.
+    };
 
-    maybe_action_names
+    let entity = siren::Entity::default()
+        .with_class_member(http_api::PATH) // FIXME: Is this 'swaps' (http_api::PATH) correct?
+        .with_properties(swap)
+        // TODO: Add the entities?
```

Comment:
Yes!

Reply:
I've pushed a bunch of code that starts with the properties.